internal/pkg/instrumentation/manager.go

```diff
@@ -103,9 +103,11 @@ func (m *Manager) FilterUnusedProbes(target *process.TargetDetails) {
 	for name, inst := range m.probes {
 		funcsFound := false
 		for _, s := range inst.Manifest().Symbols {
-			if _, exists := existingFuncMap[s]; exists {
-				funcsFound = true
-				break
+			if len(s.DependsOn) == 0 {
```

Comment:
Shouldn't we look at the contents of `s.DependsOn` to make sure the required function is present? maybe I missed that

Reply:
I don't think we need to. If DependsOn has something, it means that the probe is not valid as a standalone and there's one or more other probes that drive its functionality. So to activate this bpf program, there must be at least one found standalone probe which doesn't depend on anything else.

I added a test for this at line 85 in the manager_test.go. 
```
t.Run("HTTP server and client dependent function only target details"
```

Technically the field can be a boolean, but after a discussion a few weeks back at the SIG it was suggested that I make this field an array.

Perhaps one thing I can do is ensure that the dependents listed in DependsOn exist, so that we don't accidentally break functionality.

Reply:
Added dependent validation and tests.